src/Tools/dotnet-dump/Program.cs

```diff
@@ -49,31 +63,69 @@ public static Task<int> Main(string[] args)
+                Argument = new Argument<string>(name: "diagnosticPort")
             };
 
         private static Option OutputOption() =>
             new(
                 aliases: new[] { "-o", "--output" },
-                description: @"The path where collected dumps should be written. Defaults to '.\dump_YYYYMMDD_HHMMSS.dmp' on Windows and './core_YYYYMMDD_HHMMSS' 
+                description: @"The path where collected dumps should be written. Defaults to '.\dump_YYYYMMDD_HHMMSS.dmp' on Windows and './core_YYYYMMDD_HHMMSS'
 on Linux where YYYYMMDD is Year/Month/Day and HHMMSS is Hour/Minute/Second. Otherwise, it is the full path and file name of the dump.")
             {
+                Name = nameof(DumpCollectionConfig.DumpOutputPath),
                 Argument = new Argument<string>(name: "output_dump_path")
             };
 
         private static Option DiagnosticLoggingOption() =>
             new(
-                alias: "--diag",
+                alias: "--log",
                 description: "Enable dump collection diagnostic logging.")
             {
-                Argument = new Argument<bool>(name: "diag")
+                Name = nameof(DumpCollectionConfig.LogLevel),
+                Argument = new Argument<Dumper.LogLevelOption>(name: "log", getDefaultValue: static () => Dumper.LogLevelOption.None)
+            };
+
+        // Until we update system.commandline to a newer version, the semmantics of this are... not ideal:
+        // if the user wants the log to go to file they either have to use logToFile for the default value,
+        // or they have to use logFilePath.
+        // See https://github.com/dotnet/command-line-api/pull/1462
+        private static Option<bool> FileLoggingOption() =>
+            new(
+                alias: "--log-to-file",
+                description: "Route diagnostic logging to a file.",
+                getDefaultValue: static () => false)
+            {
+                Name = nameof(DumpCollectionConfig.LogToFile),
+                Argument = new Argument<bool>(name: nameof(DumpCollectionConfig.LogToFile)),
+                IsRequired = false
+            };
+
+        private static Option<string> LoggingPathOption() =>
+            new(
+                alias: "--log-file-path",
+                description: "Route diagnostic logging to a specific file.")
+            {
+                Name = nameof(DumpCollectionConfig.DiagnosticLogPath),
+                Argument = new Argument<string>(name: nameof(DumpCollectionConfig.DiagnosticLogPath)),
+                IsRequired = false
             };
 
         private static Option CrashReportOption() =>
             new(
                 alias: "--crashreport",
                 description: "Enable crash report generation.")
             {
+                Name = nameof(DumpCollectionConfig.GenerateCrashReport),
                 Argument = new Argument<bool>(name: "crashreport")
```

Comment:
Not consistent with using nameof(DumpCollectionConfig.GenerateCrashReport) here, but it is ok if it affects the --help text.